### DISCUSSION TITLE
next/image not adding width and height attributes when using local image

### DISCUSSION THREAD

**Discussion by runoncedev** [vercel/next.js#38219] [Help]
I'm statically importing a PNG file and passing it to the `<Image />` component:

```javascript
import Image from "next/image";

import logo from "../public/logo.png";

<Image src={logo} alt="Logo" />
```

The image is displayed but the rendered `<image />` element does not have height or width attributes set.

I'm following this documentation:

* https://nextjs.org/docs/basic-features/image-optimization#local-images

What I'm missing?

Using `next@12`.

**Comment by Icegreeen** [comment]
Have you tried setting the width and height attributes on the component?

These are the default values as the documentation says:

width={500}
height={500}

 Try changing the values

**Reply by runoncedev** [reply]
Yes, I also tried passing this values explicitly but the `<image>` still doesn't have the width and height attributes. The image itself looks good by Lighthouse complains about the lack of this attributes.

**Comment by Babbili** [comment]
@oncet you can set `height` and `width` properties in the `<Image />` component as in [Image Component API](https://nextjs.org/docs/api-reference/next/image)
```
import Image from "next/image";

import logo from "../public/logo.png";

<Image src={logo} width={300} height={400} alt="Logo" />  //  width or height here represents original value in pixels
```
There are also other properties you can make use of in the API, such as [loader](https://nextjs.org/docs/api-reference/next/image#loader), [layout](https://nextjs.org/docs/api-reference/next/image#layout) ex: `layout='responsive'` , [quality](https://nextjs.org/docs/api-reference/next/image#quality) ex: `quality='100'` an integer between 1 and 100 where 100 is the best quality. Defaults to 75

**Reply by runoncedev** [reply]
According to the documentation, shouldn't this props be provided automatically?

From the docs:

```
<Image
        src={profilePic}
        alt="Picture of the author"
        // width={500} automatically provided
        // height={500} automatically provided
        // blurDataURL="data:..." automatically provided
        // placeholder="blur" // Optional blur-up while loading
      />
```

**Reply by Babbili** [reply]
@oncet you need to provide these properties to what suits your needs, they are not provided,
According to [next/image docs](https://nextjs.org/docs/api-reference/next/image) `src`, `width` and `height` are required, others are optional.

For the `lighthouse audit` i'm getting the same as well, i made a separate discussion about it #38238

**Reply by runoncedev** [reply]
Maybe I'm not understanding the documentation right, what does it mean that width and height are "automatically provided" then? 🤔
